editor/components/inserter/menu.js

```diff
@@ -12,11 +13,59 @@ import Dashicon from 'components/dashicon';
 class InserterMenu extends wp.element.Component {
 	constructor() {
 		super( ...arguments );
+		this.blockTypes = wp.blocks.getBlocks();
+		this.categories = wp.blocks.getCategories();
+		this.nodes = {};
 		this.state = {
-			filterValue: ''
+			filterValue: '',
+			currentFocus: null
 		};
 		this.filter = this.filter.bind( this );
 		this.instanceId = this.constructor.instances++;
+		this.isShownBlock = this.isShownBlock.bind( this );
+		this.changeMenuSelection = this.changeMenuSelection.bind( this );
+		this.setSearchFocus = this.setSearchFocus.bind( this );
+		this.bindReferenceNode = this.bindReferenceNode.bind( this );
+		this.onKeyDown = this.onKeyDown.bind( this );
+		this.getVisibleBlocks = this.getVisibleBlocks.bind( this );
+		this.getVisibleBlocksByCategory = this.getVisibleBlocksByCategory.bind( this );
+		this.getCategoryIndex = this.getCategoryIndex.bind( this );
+		this.sortBlocksByCategory = this.sortBlocksByCategory.bind( this );
```

Comment:
It looks like `changeMenuSelection`, `bindReferenceNode`, `getVisibleBlocksByCategory`, `getCategoryIndex` don't require binding?